flow-server/src/main/java/com/vaadin/flow/server/SynchronizedRequestHandler.java

```diff
@@ -58,13 +84,60 @@ public boolean handleRequest(VaadinSession session, VaadinRequest request,
      *            The response object to which a response can be written.
      * @return true if a response has been written and no further request
      *         handlers should be called, otherwise false
-     *
      * @throws IOException
      *             If an IO error occurred
+     * @see #handleRequest(VaadinSession, VaadinRequest, VaadinResponse)
      */
     public abstract boolean synchronizedHandleRequest(VaadinSession session,
             VaadinRequest request, VaadinResponse response) throws IOException;
 
+    /**
+     * Returns {@literal true} if
+     * {@link #synchronizedHandleRequest(VaadinSession, VaadinRequest, VaadinResponse, String)}
+     * should be called. Returns {@literal false} if
+     * {@link #synchronizedHandleRequest(VaadinSession, VaadinRequest, VaadinResponse)}
+     * should be called.
```

Comment:
I would make "Gets if request body should be read before calling synchronizedHandleRequest." as initial documentation, to make immediately clear the effect of the flag.

```suggestion
     * Gets if request body should be read before calling synchronizedHandleRequest.
     *<p></p>
     * Returns {@literal true} if
     * {@link #synchronizedHandleRequest(VaadinSession, VaadinRequest, VaadinResponse, String)}
     * should be called. Returns {@literal false} if
     * {@link #synchronizedHandleRequest(VaadinSession, VaadinRequest, VaadinResponse)}
     * should be called.
```